### DISCUSSION TITLE
Using extend-footer.html for custom balance-text.js script

### DISCUSSION THREAD

**Discussion by manchuwook** [nunocoracao/blowfish#596] [Q&A] [answered]
Hello, I see in the template that there is an option to do `extend-footer.html` where I can add a reference to a custom javascript ([Adobe balance-text](https://github.com/adobe/balance-text))
````handlebars
  {{/* Extend footer - eg. for extra scripts, etc. */}}
  {{ if templates.Exists "partials/extend-footer.html" }}
  {{ partialCached "extend-footer.html" . }}
  {{ end }}
````
But I get the error `.File.Path on zero object. Wrap it in if or with: {{ with .File }}{{ .Path }}{{ end }}`

Is there something special I need to get this to work?  I need to add something to the .prose class for my custom.css in order to activate it.

I tried adding the js to `<rootDir>static/js/balance.js` with the extend-footer in `<rootDir>layouts/partials/extend-footer.html`.  I will likely also need to add the balance-text global to the head section of the template as well, possibly through a CDN - though I'd rather store a local copy.

**Comment by nunocoracao** [comment, marked as answer]
Weird, that should work. 
Overall using the extend-footer seems like the best approach. That static version should work if you are just importing the JS using normal HTML. If you want to optimize it you can place it in assets and use a similar logic to what's in the head file to import it using Hugo itself. 
Hope this helped

**Comment by manchuwook** [comment]
It did, I apparently had a floating index.md that had no front matter to it.  I can't get my backgrounds to show up in articles - which would be a different topic.  I'm reading up on that.
